File: DeepLearningFromZero/2.perceptron/perceptron.py
# -*- coding: utf-8 -*-

def calc(w1, w2, bias, x1, x2):
    tmp = bias + w1 * x1 + w2 * x2
    if (tmp >= 0):
        return 1
    else:
        return 0

def AND(x1, x2):
    return calc(0.5, 0.5, -0.9, x1, x2)

def OR(x1, x2):
    return calc(0.5, 0.5, -0.5, x1, x2)

def NAND(x1, x2):
    return calc(-0.5, -0.5, 0.9, x1, x2)
        
def XOR(x1, x2):
    return AND(OR(x1, x2), NAND(x1, x2))        
        
print(AND(0, 0));
print(AND(0, 1));
print(AND(1, 0));
print(AND(1, 1))
print("")

print(OR(0, 0));
print(OR(0, 1));
print(OR(1, 0));
print(OR(1, 1));
print("")

print(NAND(0, 0));
print(NAND(0, 1));
print(NAND(1, 0));
print(NAND(1, 1));
print("")

print(XOR(0, 0));
print(XOR(0, 1));
print(XOR(1, 0));
print(XOR(1, 1));
print("")

File: DeepLearningFromZeroKot/src/c2perceptron/perceptron.kt
package c2perceptron

fun calc(w1: Double, w2: Double, bias: Double, x1: Double, x2: Double): Double {
    val tmp = bias + w1 * x1 + w2 * x2
    return if (tmp >= 0)
        1.0
    else
        0.0
}

fun and(x1: Double, x2: Double): Double {
    return calc(0.5, 0.5, -0.9, x1, x2)
}

fun or(x1: Double, x2: Double): Double {
    return calc(0.5, 0.5, -0.5, x1, x2)
}

fun nand(x1: Double, x2: Double): Double {
    return calc(-0.5, -0.5, 0.9, x1, x2)
}

fun xor(x1: Double, x2: Double): Double {
    return and(or(x1, x2), nand(x1, x2))
}

fun main() {
    println(and(0.0, 0.0))
    println(and(0.0, 1.0))
    println(and(1.0, 0.0))
    println(and(1.0, 1.0))
    println("")

    println(or(0.0, 0.0))
    println(or(0.0, 1.0))
    println(or(1.0, 0.0))
    println(or(1.0, 1.0))
    print("")

    println(nand(0.0, 0.0))
    println(nand(0.0, 1.0))
    println(nand(1.0, 0.0))
    println(nand(1.0, 1.0))
    println("")

    println(xor(0.0, 0.0))
    println(xor(0.0, 1.0))
    println(xor(1.0, 0.0))
    println(xor(1.0, 1.0))
    println("")
}

